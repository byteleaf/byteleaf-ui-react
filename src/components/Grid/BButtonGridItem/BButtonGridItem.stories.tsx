import React from 'react';
import { Button } from '@material-ui/core';

import { BButtonGridItem } from './BButtonGridItem';

export default {
  title: 'Grid|BButtonGridItem',
};

export const BButtonGridItemStory = () => (
  <BButtonGridItem>
    <Button>Submit</Button>
  </BButtonGridItem>
);

BButtonGridItemStory.story = {
  name: 'default',
  parameters: {
    notes: 'A grid item for wrapping e.g. a submit button. Centers the button and adds a small margin to the top.',
  },
};
