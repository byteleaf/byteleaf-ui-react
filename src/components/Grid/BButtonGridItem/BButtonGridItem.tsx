import { Grid, makeStyles } from '@material-ui/core';
import React from 'react';
import theme from '../../../theme';

const useStyles = makeStyles({
  root: {
    paddingTop: theme.spacing(4),
    textAlign: 'center',
  },
});

export const BButtonGridItem = ({ children }: { children: JSX.Element | JSX.Element[] }) => {
  const classes = useStyles();

  return (
    <Grid item xs={12} className={classes.root}>
      {children}
    </Grid>
  );
};
