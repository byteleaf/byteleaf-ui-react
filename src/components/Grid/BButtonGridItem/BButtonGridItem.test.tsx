import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BButtonGridItem } from './BButtonGridItem';

describe('BButtonGridItem', () => {
  it('renders correctly', async () => {
    const { container } = render(
      <BButtonGridItem>
        <button>test</button>
      </BButtonGridItem>,
    );

    expect(container).toMatchSnapshot();
  });
});
