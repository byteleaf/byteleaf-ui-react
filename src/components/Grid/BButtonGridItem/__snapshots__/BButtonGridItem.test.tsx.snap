// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BButtonGridItem renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <button>
      test
    </button>
  </div>
</div>
`;
